### PR TITLE
Document link_section arbitrary bytes

### LINKED ISSUE #60633: `#[link_section] static FOO` can contain undefined bytes on wasm
https://github.com/rust-lang/rust/blob/e80c86c535b74cdd65c4b75b94ce38342d0bb946/src/librustc_typeck/check/mod.rs#L1444-L1467 only checks that the link section contains no pointers, but there may be arbitrary bytes in the data, even undefined bytes (e.g. due to padding).

cc @alexcrichton is that ok? llvm may insert `undef` whereever there are undefined bytes. Or should we lint/error on such statics?

This issue has been assigned to @chocol4te via [this comment](https://github.com/rust-lang/rust/issues/60633#issuecomment-512236594).

### PR DESCRIPTION
**Pull request by fmckeogh**
Closes #60633.

### CODE REVIEW

File: src/librustc_typeck/check/mod.rs [hunk 1]
```diff
@@ -1447,6 +1447,9 @@ fn maybe_check_static_with_link_section(tcx: TyCtxt<'_>, id: DefId, span: Span)
     // other executable formats. Namely we can only embed a list of bytes,
     // nothing with pointers to anything else or relocations. If any relocation
     // show up, reject them here.
+    // #[link_section] may contain arbitrary, or even undefined bytes, but it is
```
Comment (estebank) [inline src/librustc_typeck/check/mod.rs]:
```suggestion
    // `#[link_section]` may contain arbitrary, or even undefined bytes, but it is
```

**Comment by estebank** [comment]
r=me with the nitpick fixed

**Comment by chocol4te** [comment]
r?=estebank

**Pull request merged**